### PR TITLE
Addition of the XSD 1.1 validator to allow for Conditional Type Assignment

### LINKED ISSUE #1500: [TASK] Support for validation XSD version 1.1
--------
Issue Description
--------

**Is your feature request related to a problem? Please describe.**
The XSD 1.1 allows for Conditional Type Assignment (see https://www.w3.org/TR/xmlschema11-1/#cTypeAlternative).
This capability will allow RAVEN input file to be validate at 90% using an XSD validator. This will reduce enormously
the runtime errors (caused by input files not conforming)  

**Describe the solution you'd like**
XSD 1.1 validator

**Describe alternatives you've considered**
None

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### LINKED ISSUE #1502: [TASK] XSD checking moving to 1.1 version
--------
Issue Description
--------

**Is your feature request related to a problem? Please describe.**
Moving RAVEN validation XSD to 1.1 for conditional type checking

The XSD 1.1 allows for Conditional Type Assignment (see https://www.w3.org/TR/xmlschema11-1/#cTypeAlternative).
This capability will allow RAVEN input file to be validate at 90% using an XSD validator. This will reduce enormously
the runtime errors (caused by input files not conforming)

**Describe the solution you'd like**
 
**Describe alternatives you've considered**
 

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by alfoa**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
Closes #1500
close #1502 

##### What are the significant changes in functionality due to this change request?

- Addition of the library  ```xmlschema``` that allows for validation of XSD version 1.1. : 
- In RAVEN we need ```Conditional Type Assigment``` for checking the RAVEN input file before even running the software.
See https://stackoverflow.com/questions/27878402/how-to-make-type-depend-on-attribute-value-using-conditional-type-assignment
- The ```xmlint``` usage has been replaced by the new library and the validator script updated.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

**Comment by PaulTalbot-INL** [comment]
The documentation (https://www.w3.org/TR/xmlschema11-1/#cTypeAlternative) seems fairly painful to parse without being an expert, but hopefully we can make it do what we want. It would be nice to see that this works for what we need before adding the dependency; can we see it in our XSD?

File: developer_tools/validate_xml.py [hunk 1]
```diff
@@ -32,6 +33,8 @@ def validateTests():
   """
   print('Beginning test validation...')
   tests = get_coverage_tests.getRegressionTests(skipExpectedFails=True)
+  schema = os.path.join(scriptDir, 'XSDSchemas', 'raven.xsd')
+  my_schema = xmlschema.XMLSchema11(schema)
```
Comment (PaulTalbot-INL) [inline developer_tools/validate_xml.py:37]: this should be `mySchema`

**Review by PaulTalbot-INL — CHANGES_REQUESTED** [review]
Only one non-conforming variable to fix.

File: developer_tools/validate_xml.py (hunk 1 above)
Comment (wangcj05) [reply developer_tools/validate_xml.py]: changed.

**Comment by wangcj05** [comment]
I have updated this branch, and it seems it works well. @PaulTalbot-INL @joshua-cogliati-inl @mandd do you have any comments?

**Comment by joshua-cogliati-inl** [comment]
I have no objections to switching to a different validator (with the comment that we really need to finish getting InputData used everywhere).

**Comment by wangcj05** [comment]
As we discussed today, I think we all agreed we can update XSD to 1.1, and we will push the fully deployment of InputData for RAVEN. @PaulTalbot-INL do you have any comments for this PR? I saw you have reviewed it in the past.

**Review by wangcj05 — APPROVED** [review]

**Review by mandd — APPROVED** [review]

**Comment by wangcj05** [comment]
Checklist is satisfied, and PR have been approved by another developer. This PR can be merged.

**Pull request merged**